src/core/observables/ChainAngles.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright (C) 2019 The ESPResSo project
+
+This file is part of ESPResSo.
+
+ESPResSo is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+ESPResSo is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef OBSERVABLES_CHAINANGLES_HPP
+#define OBSERVABLES_CHAINANGLES_HPP
+
+#include "PidObservable.hpp"
+#include <utils/Vector.hpp>
+
+#include <cmath>
+#include <vector>
+
+namespace Observables {
+
+/** Calculate bond angles between particles in a polymer.
+ *  For @f$ n @f$ bonded particles, return the @f$ n-2 @f$ angles along the
+ *  chain, in radians.
```

Comment:
copy-paste error: you're actually calculating `n-2` angles averages between particles that are `n`-bond apart